service/health/searcher/service.go

```diff
@@ -51,13 +53,84 @@ func New(config Config) (*Service, error) {
 	return s, nil
 }
 
+func extractStatusCluster(status v1alpha1.StatusCluster) ClusterInfo {
+	nodes := []NodeInfo{}
+	for _, node := range status.Nodes {
+		nodes = append(nodes, NodeInfo{
+			labels: node.Labels,
+		})
+	}
+	return ClusterInfo{
+		nodes,
+	}
+}
+
+// searchAWSCR searches for the cluster config in AWSClusterConfigs resources.
+func (s *Service) searchAWSInfo(ctx context.Context, clusterID string) (*ClusterInfo, error) {
+	cr, err := s.g8sClient.ProviderV1alpha1().AWSConfigs("default").Get(clusterID, v1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterInfo := extractStatusCluster(cr.Status.Cluster)
+	return &clusterInfo, nil
+}
+
+// searchAzureCR searches for the cluster config in AWSClusterConfigs resources.
+func (s *Service) searchAzureInfo(ctx context.Context, clusterID string) (*ClusterInfo, error) {
+	cr, err := s.g8sClient.ProviderV1alpha1().AzureConfigs("default").Get(clusterID, v1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterInfo := extractStatusCluster(cr.Status.Cluster)
+	return &clusterInfo, nil
+}
+
+// searchKVMCR abc
+func (s *Service) searchKVMInfo(ctx context.Context, clusterID string) (*ClusterInfo, error) {
+	cr, err := s.g8sClient.ProviderV1alpha1().KVMConfigs("default").Get(clusterID, v1.GetOptions{})
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	clusterInfo := extractStatusCluster(cr.Status.Cluster)
+	return &clusterInfo, nil
+}
+
+func evaluateGeneralHealth(cluster *ClusterInfo) GeneralStatus {
```

Comment:
I would call it `NewGeneralStatus` and put it together with the type definition where the constructors belong.

Reply:
I didn't see it as a constructor, but now I see what you mean. Done.